airflow/gcp/hooks/pubsub.py

```diff
@@ -219,30 +383,50 @@ def delete_subscription(self, project: str, subscription: str, fail_if_not_exist
         :param subscription: the Pub/Sub subscription name to delete; do not
             include the ``projects/{project}/subscriptions/`` prefix.
         :type subscription: str
-        :param fail_if_not_exists: if set, raise an exception if the topic
-            does not exist
+        :param fail_if_not_exists: if set, raise an exception if the topic does not exist
         :type fail_if_not_exists: bool
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: Sequence[Tuple[str, str]]]
         """
-        service = self.get_conn()
-        full_subscription = _format_subscription(project, subscription)
+        subscriber = self.subscriber_client
+        subscription_path = SubscriberClient.subscription_path(project, subscription)  # noqa E501 # pylint: disable=no-member,line-too-long
+
+        self.log.info("Deleting subscription (path) %s", subscription_path)
         try:
-            service.projects().subscriptions().delete(  # pylint: disable=no-member
-                subscription=full_subscription).execute(num_retries=self.num_retries)
-        except HttpError as e:
-            # Status code 404 indicates that the subscription was not found
-            if str(e.resp['status']) == '404':
-                message = 'Subscription does not exist: {}'.format(
-                    full_subscription)
-                self.log.warning(message)
-                if fail_if_not_exists:
-                    raise PubSubException(message)
-            else:
-                raise PubSubException(
-                    'Error deleting subscription {}'.format(full_subscription),
-                    e)
+            # pylint: disable=no-member
+            subscriber.delete_subscription(
+                subscription=subscription_path,
+                retry=retry,
+                timeout=timeout,
+                metadata=metadata
+            )
+
+        except NotFound:
+            message = 'Subscription does not exist: {}'.format(subscription_path)
+            self.log.warning(message)
+            if fail_if_not_exists:
+                raise PubSubException(message)
+        except GoogleAPICallError as e:
```

Comment:
This part of code is not covered by unit test. Can you complete missing unit tests?

Reply:
I've added test.